blockchain/chain_sync/src/sync.rs

```diff
@@ -193,6 +195,40 @@ where
                             )
                             .await
                     }
+                    Some(NetworkEvent::PubsubMessage { source, message }) => {
+                        match message {
+                            forest_libp2p::PubsubMessage::Block(b) => {
+                                info!("Received block over GossipSub: {} from {}", b.header.epoch(), source.clone().unwrap());
+                                // Get bls_messages in the store or over Bitswap
+                                let bmsgs: Vec<_> = b.bls_messages.into_iter().map(|m| self.network.bitswap_get::<UnsignedMessage>(m)).collect();
+                                let bmsgs = try_join_all(bmsgs).await;
+                                if let Err(e) = &bmsgs {
+                                    warn!("Failed to get UnsignedMessage: {}", e);
+                                    ()
+                                }
```

Comment:
join after you send the signed message requests out, no reason to await all of these to get the secp ones